api/tokens_test.go

```diff
@@ -134,3 +140,85 @@ func TestHandleUnexpectedError(t *testing.T) {
 		So(resp.Body.String(), ShouldResemble, errorResponseBodyExample)
 	})
 }
+
+func TestSignOutHandler(t *testing.T) {
+	var (
+		r                                                  = mux.NewRouter()
+		ctx                                                = context.Background()
+		requestType, poolId, clientId, clientSecret string = "DELETE", "us-west-11_bxushuds", "client-aaa-bbb", "secret-ccc-ddd"
+	)
+
+	m := &mock.MockCognitoIdentityProviderClient{}
+
+	// mock call to: GlobalSignOut(input *cognitoidentityprovider.GlobalSignOutInput) (*cognitoidentityprovider.GlobalSignOutOutput, error)
+	m.GlobalSignOutFunc = func(signOutInput *cognitoidentityprovider.GlobalSignOutInput) (*cognitoidentityprovider.GlobalSignOutOutput, error) {
+		return &cognitoidentityprovider.GlobalSignOutOutput{}, nil
+	}
+
+	api := Setup(ctx, r, m, poolId, clientId, clientSecret)
+
+	Convey("Global Sign Out returns 204: successfully signed out user", t, func() {
+		r := httptest.NewRequest(requestType, signOutEndPoint, bytes.NewReader(nil))
+		r.Header.Set("Authorization", "Bearer zzzz-yyyy-xxxx")
+
+		w := httptest.NewRecorder()
+
+		api.Router.ServeHTTP(w, r)
+
+		So(w.Code, ShouldEqual, http.StatusNoContent)
+	})
+
+	Convey("Global Sign Out returns 400: malformed header", t, func() {
+		r := httptest.NewRequest(requestType, signOutEndPoint, bytes.NewReader(nil))
+		r.Header.Set("Authorization", "Bearerzzzz-yyyy-xxxx")
+
+		w := httptest.NewRecorder()
+
+		api.Router.ServeHTTP(w, r)
+
+		So(w.Code, ShouldEqual, http.StatusBadRequest)
+	})
+
+	Convey("Global Sign Out returns 400: no header value", t, func() {
+		r := httptest.NewRequest(requestType, signOutEndPoint, bytes.NewReader(nil))
+		r.Header.Set("Authorization", "")
+
+		w := httptest.NewRecorder()
+
+		api.Router.ServeHTTP(w, r)
+
+		So(w.Code, ShouldEqual, http.StatusBadRequest)
+	})
+
+	Convey("Global Sign Out returns 500: Cognito internal error", t, func() {
+		r := httptest.NewRequest(requestType, signOutEndPoint, bytes.NewReader(nil))
+		r.Header.Set("Authorization", "Bearer zzzz-yyyy-xxxx")
+
+		// mock failed call to: GlobalSignOut(input *cognitoidentityprovider.GlobalSignOutInput) (*cognitoidentityprovider.GlobalSignOutOutput, error)
+		m.GlobalSignOutFunc = func(signOutInput *cognitoidentityprovider.GlobalSignOutInput) (*cognitoidentityprovider.GlobalSignOutOutput, error) {
+			return nil, errors.New("InternalErrorException: Something went wrong")
+		}
+
+		w := httptest.NewRecorder()
+
+		api.Router.ServeHTTP(w, r)
+
+		So(w.Code, ShouldEqual, http.StatusInternalServerError)
+	})
+
+	Convey("Global Sign Out returns 400: request error", t, func() {
```

Comment:
Could combine all `400` tests by putting all test scenarios in a struct and iterating over them using:

`for _, tt := range <STRUCT_NAME> { ... }`